datumaro/components/converter.py

```diff
@@ -34,8 +36,22 @@ def convert(cls, extractor, save_dir, **options):
         return converter.apply()
 
     @classmethod
+    @error_rollback
     def patch(cls, dataset, patch, save_dir, **options):
-        return cls.convert(dataset, save_dir, **options)
+        # Is not any better in performance than just writing a dataset,
+        # but in case of patching (i.e. writing to the previous location),
+        # it allows to avoid many problems with removing and replacing
+        # existing files. Surely, this approach also has problems with
```

Comment:
> For example, what if the user tries to save one dataset into the directory of another dataset?

I'd say it's not our business. But, for completeness sake, there is always a check for empty directory and a corresponding option for overriding in the CLI.

Reply:
> I'd say it's not our business.

Why not? Don't you think it's inconsistent that saving a dataset in its original directory wipes that directory, but saving it in another directory doesn't?

Reply:
Potentially, there can be extra files in the dataset directory (like readmes, descriptions and stuff), so I don't think removing the directory in patching is also a good solution. But it at least something working and doable.

Reply:
I realize that, but I think the behavior should be consistent between scenarios. I think it would be surprising for the user to see that some write operations leave the existing files as-is, while some wipe them out.

Perhaps the right solution would be for the user to opt-in to the wipe behavior. You could add a command-line switch, and in the absence of that switch refuse to patch unless the dataset has a specialized patch implementation.

Reply:
Adding an extra mystical cli parameter doesn't look right for me. I propose to leave this question till we face this problem IRL.